packages/upload-api/src/ucan/conclude.js

```diff
@@ -0,0 +1,163 @@
+    const ranInvocation = receipt.ran
+    const httpPutTaskGetRes = await tasksStorage.get(ranInvocation.link())
+    if (httpPutTaskGetRes.error) {
+      return httpPutTaskGetRes
+    }
+
+    // Store receipt
+    const receiptPutRes = await receiptsStorage.put(receipt)
+    if (receiptPutRes.error) {
+      return {
+        error: receiptPutRes.error,
+      }
+    }
+
+    // THIS IS A TEMPORARY HACK
+    // Schedule `blob/accept` if there is a `http/put` capabilities
+    // inside the invocation that this receipt comes from
+    const scheduleRes = await Promise.all(
+      httpPutTaskGetRes.ok.capabilities
+        // Go through invocation tasks and get all `http/put`
+        .filter((cap) => cap.can === HTTP.put.can)
+        // @ts-expect-error body exists in `http/put` but unknown type here
+        .map(async (/** @type {API.HTTPPut} */ cap) => {
+          // Get triggering task (blob/allocate) by checking blocking task from `url`
+          /** @type {API.UnknownLink} */
+          // @ts-expect-error ts does not know how to get this
+          const blobAllocateTaskCid = cap.nb.url['ucan/await'][1]
+          const blobAllocateTaskGet = await tasksStorage.get(blobAllocateTaskCid)
+          if (blobAllocateTaskGet.error) {
+            return blobAllocateTaskGet
+          }
+
+          /** @type {API.BlobAllocate} */
+          // @ts-expect-error ts does not know how to get this
+          const allocateCapability = blobAllocateTaskGet.ok.capabilities.find(
+            // @ts-expect-error ts does not know how to get this
+            (/** @type {API.BlobAllocate} */ allocateCap) =>
+              equals(allocateCap.nb.blob.digest, cap.nb.body.digest) && allocateCap.can === W3sBlob.allocate.can
+          )
+          
+          const blobAccept = await W3sBlob.accept
+            .invoke({
+              issuer: id,
+              audience: id,
+              with: id.toDIDKey(),
+              nb: {
+                blob: cap.nb.body,
+                exp: Number.MAX_SAFE_INTEGER,
+                space: allocateCapability.nb.space,
+                _put: {
+                  'ucan/await': ['.out.ok', ranInvocation.link()],
+                },
+              },
+              expiration: Infinity,
+            })
+            .delegate()
+
+          return tasksScheduler.schedule(blobAccept)
+        })
+    )
+
+    const scheduleErrors = scheduleRes.filter((res) => res.error)
+    if (scheduleErrors.length && scheduleErrors[0].error) {
+      return {
+        error: scheduleErrors[0].error,
+      }
+    }
+
+    return {
+      ok: { time: Date.now() },
+    }
+  })
+
+/**
+ * @param {import('multiformats').UnknownLink} cid
+ * @param {IterableIterator<import('@ucanto/interface').Block<unknown, number, number, 1>>} blocks
+ * @returns {Promise<import('@ucanto/interface').Result<Uint8Array, DecodeBlockOperationFailed>>}
+ */
+export const findBlock = async (cid, blocks) => {
```

Comment:
I can't see this function used anywhere, do we need it ?

Reply:
old thing, I will remove